### PR TITLE
Fix 0.6 depwarns

### PR DESCRIPTION
**Pull request by yuyichao**
Test on master might fail with an error similar to

```
signal (4): Illegal instruction
while loading /home/yuyichao/projects/contrib/PyCall.jl/test/runtests.jl, in expression starting on line 318
copy at /usr/share/julia/site/v0.6/PyCall/src/numpy.jl:340
unknown function (ip: 0x7fe154c964f2)
jl_call_method_internal at /build/julia-git/src/julia-avx2/src/julia_internal.h:248
```

and that's https://github.com/JuliaLang/julia/issues/20305

I also decided to work around https://github.com/JuliaLang/julia/issues/19987 since it doesn't seems to have very high priority or trivial to fix (but trivial to workaround)....

### CODE REVIEW

**Comment by yuyichao** [comment]
The CI failure on 0.4 is caused by using the conditional comprehension syntax which exists on master.

**Comment by yuyichao** [comment]
0.4 syntax error fixed.

**Comment by yuyichao** [comment]
So the error doesn't occur on travis..... Maybe because numpy isn't installed? (It was triggered in a numpy specific test...)

File: src/conversions.jl [hunk 1]
```diff
@@ -140,9 +140,10 @@ pyptr_query(po::PyObject) = pyisinstance(po, c_void_p_Type) || pyisinstance(po,
 # typealias PyAny Union{PyObject, Int, Bool, Float64, Complex128, AbstractString, Function, Dict, Tuple, Array}
 abstract PyAny
 
-pyany_toany(T::Type) = T
-pyany_toany(T::Type{PyAny}) = Any
-pyany_toany(T::Type{Vararg{PyAny}}) = Vararg{Any}
+function pyany_toany(T::Type)
+    T === Vararg{PyAny} ? Vararg{Any} : T
+end
+pyany_toany(::Type{PyAny}) = Any
```
Comment (stevengj) [inline src/conversions.jl]: Why was this changed?

Comment (yuyichao) [inline src/conversions.jl]: `Type{Vararg{PyAny}}` isn't a valid type anymore.

Comment (stevengj) [inline src/conversions.jl]: Why not?

Comment (yuyichao) [inline src/conversions.jl]: I don't know the full story but it was never an actual type (you can't find a value that matches this type)

Comment (stevengj) [inline src/conversions.jl]: I'm confused, so why does `T === Vararg{PyAny}` still work if that is not a type?

Comment (yuyichao) [inline src/conversions.jl]: The usage of it has always been strange. It is essentially only meaningful in a tuple but it's hard to only allow it there without breaking many other things. It is used to be allowed to use it just like a type though almost all types constructed this way are invalid. This is not disallowed.

It is still allowed to be used as a value and is still treated as a type for dispatch purpose even if it never actually was one. I believe the long term goal is to get rid of it or somehow being able to pass it around without dealing with this special case or generating invalid types.

**Comment by malmaud** [comment]
@stevengj Any objection to tagging PyCall now that we have this?

**Comment by stevengj** [comment]
Tagged 1.9
